doc/doxygen/doxyfile.in

```diff
@@ -280,25 +310,27 @@ PREDEFINED =           = "HAVE_QHULL=1" \
                          "HAVE_DAVIDSDK=1" \
                          "HAVE_DSSDK=1" \
                          "HAVE_RSSDK=1" \
-                         "DOXYGEN_ONLY=1"
+                         "DOXYGEN_ONLY=1" \
+                         "_WIN32=1" \
+                         "PCL_DEPRECATED(x)=/** \deprecated x */"
```

Comment:
Will this not double generate documentation? I saw that you've added `\deprecated` to `pcl::int8_t` and others.

Reply:
Without the doxyfile.in edit I get
```
/opt/libpcl/pcl/common/include/pcl/common/io.h:335: warning: Found ';' while parsing initializer list! (doxygen could be confused by a macro call without semicolon)

...

/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:92: warning: documented symbol `typedef std::uint8_t pcl::uint8_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:93: warning: documented symbol `typedef std::int8_t pcl::int8_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:94: warning: documented symbol `typedef std::uint16_t pcl::uint16_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:95: warning: documented symbol `typedef std::int16_t pcl::int16_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:96: warning: documented symbol `typedef std::uint32_t pcl::uint32_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:97: warning: documented symbol `typedef std::int32_t pcl::int32_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:98: warning: documented symbol `typedef std::uint64_t pcl::uint64_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:99: warning: documented symbol `typedef std::int64_t pcl::int64_t' was not declared or defined.
/opt/libpcl/pcl/common/include/pcl/pcl_macros.h:100: warning: documented symbol `typedef std::int_fast16_t pcl::int_fast16_t' was not declared or defined.

...

/opt/libpcl/pcl/common/include/pcl/common/io.h:337: warning: argument 'cloud1' of command @param is not found in the argument list of pcl::copyPointCloud(const pcl::PCLPointCloud2 &cloud_in, const std::vector< int > &indices, pcl::PCLPointCloud2 &cloud_out)
/opt/libpcl/pcl/common/include/pcl/common/io.h:337: warning: argument 'cloud2' of command @param is not found in the argument list of pcl::copyPointCloud(const pcl::PCLPointCloud2 &cloud_in, const std::vector< int > &indices, pcl::PCLPointCloud2 &cloud_out)

...

/opt/libpcl/pcl/octree/include/pcl/octree/octree2buf_base.h:937: warning: argument 'n_arg' of command @param is not found in the argument list of pcl::octree::Octree2BufBase< LeafContainerT, BranchContainerT >::PCL_DEPRECATED("use std::log2 instead")
/opt/libpcl/pcl/octree/include/pcl/octree/octree_base.h:685: warning: argument 'n_arg' of command @param is not found in the argument list of pcl::octree::OctreeBase< LeafContainerT, BranchContainerT >::PCL_DEPRECATED("use std::log2 instead")
```

extra warnings, and without the pcl_macros.h edit I simply miss that documentation.

That's what I tried to explain in https://github.com/PointCloudLibrary/pcl/pull/3701#issuecomment-594012517

Reply:
That makes sense. You've fixed so many different errors with esoteric reasons that the explanations just slip out of the memory.